### DISCUSSION TITLE
Update not working - Garmin going to start charging?

### DISCUSSION THREAD

**Discussion by markbulk** [tcgoetz/GarminDB#110] [General]
I tried to do a database_update this morning and got the following:

```
Mac-mini:GarminDB some_user$ sh download_update_dbs.sh
Updating existing DBs by downloading from Garmin Connect
___Downloading Latest Data___
Geting activities: '/Users/some_user/HealthData/FitFiles/Activities' (30) temp /var/folders/9p/5j3w5chd2410q585389mpw_40000gn/T/tmpm1czocaq
0activities [00:00, ?activities/s]
Traceback (most recent call last):
  File "garmin.py", line 341, in <module>
  File "garmin.py", line 322, in main
  File "garmin.py", line 127, in download_data
  File "download_garmin.py", line 283, in get_activities
  File "tqdm/std.py", line 1167, in __iter__
TypeError: 'NoneType' object is not iterable
[1618] Failed to execute script garmin
```

I've attached the [garmin.log](https://github.com/tcgoetz/GarminDB/files/6057002/garmin.log) file.  Odd message showing up in the log file about payment required.  I hope this isn't a sign of things to come with Garmin!

This is potentially related to [this reddit thread](https://www.reddit.com/r/Garmin/comments/luggyk/garmin_wont_upload_activity_newsfeed_strava/)?

**Comment by tcgoetz** [comment]
This bug was fixed: https://github.com/tcgoetz/GarminDB/issues/109
Fix is on develop branch. Will be released soon.

**Reply by JulianParris** [reply]
I did notice I could set "activities" to false in the config file as a temp fix to get other entries, but am looking forward to the fixed release. Thank you for the work on this!

**Comment by markbulk** [comment]
Note that I just updated my setup (`pip install --upgrade garmindb`) and I'm still getting the original error.   (log file looks basically the same as what was posted)

That said, @JulianParris 's suggestion of setting `"activities" : false` works for me (i.e., all of the other elements download).

**Comment by tcgoetz** [comment]
The fix is in the 3.1.1 release.

**Reply by markbulk** [reply]
This is the right answer.  Unfortunately, despite upgrading to the latest version of GarminDB, when I ran the update command, it was running an earlier version (and failing on activities as described above).   In checking the verbose output of the `pip install --upgrade garmindb` command, I see that I do have version 3.1.1.

So instead of running `sh download_update_dbs.sh`, I instead ran `garmindb_cli.py --all --download --import --analyze --latest` and it worked.  The shell script was trying to run a similar command, but it was `garmin --all --download --import --analyze --latest`.

As a note to anybody else `garmin -v` will output the version, also `garmin --version` works, too.  That file was still at 2.0.0.

As a note to myself and others that might find it useful, running `which garmindb_cli.py`, it returned (MacOS 12.0.1), `/Library/Frameworks/Python.framework/Versions/3.9/bin/garmindb_cli.py`.

**Comment by tcgoetz** [comment]
garmin is the old app, pre pip install. garmin_cli.py is the new pip installed app. If you have both, I would delete the old one.
